control-plane/plugin/src/resources/snapshot.rs

```
@@ -73,23 +73,18 @@ async fn get_snapshots(
     volid: &Option<VolumeId>,
     snapid: &Option<SnapshotId>,
 ) -> Option<Vec<openapi::models::VolumeSnapshot>> {
-    // XXX: 500 is a ballpark number. Once pagination is implemented, we'll reduce this
-    // to max entries per call.
-    let mut snapshots = Vec::with_capacity(500);
     match RestClient::client()
         .snapshots_api()
         .get_volumes_snapshots(volid.as_ref(), snapid.as_ref())
         .await
     {
         Ok(snaps) => {
             let s = snaps.into_body();
-            snapshots.extend(s.entries);
+            Some(s.entries)
         }
         Err(e) => {
             println!("Failed to list volume snapshots. Error {e}");
-            return None;
+            None
```

hmm why is this returning an Option? Shouldn't we simply return an error here?

---

This is a helper function, whose return value is checked to decide whether to print table or not.
Again, I just followed existing CLI get operation semantics here :)